src/pandapipes/plotting/generic_geodata.py

```diff
@@ -44,29 +46,53 @@ def build_igraph_from_ppipes(net, junctions=None):
                           "`conda install igraph` "
                           "or from https://www.lfd.uci.edu/~gohlke/pythonlibs")
     g = ig.Graph(directed=True)
-    junction_index = net.junction.index if junctions is None else np.array(junctions)
+    junction_index = net.junction.index.to_numpy() if junctions is None else np.array(junctions)
     nr_junctions = len(junction_index)
     g.add_vertices(nr_junctions)
     g.vs["label"] = list(junction_index)
     pp_junction_mapping = dict(list(zip(junction_index, list(range(nr_junctions)))))
 
     for comp in net['component_list']:
-        if not issubclass(comp, BranchComponent):
+        tbl = comp.table_name()
+        if tbl in exclude_branch_elements:
+            continue
+        if edge_factories_override is not None and tbl in edge_factories_override:
+            edge_factories_override[tbl](net, g, pp_junction_mapping, junction_index)
+            continue
+        if not issubclass(comp, BranchComponent) or net[tbl].empty:
             continue
         fjc, tjc = comp.from_to_node_cols()
-        mask = _get_element_mask_from_nodes(net, comp.table_name(), [fjc, tjc], junctions)
-        for comp_data in net[comp.table_name()][mask].itertuples():
-            weight = 0.001 if 'length_km' not in dir(comp_data) else getattr(comp_data, 'length_km')
-            g.add_edge(pp_junction_mapping[getattr(comp_data, fjc)], pp_junction_mapping[getattr(comp_data, tjc)],
-                       weight=weight)
+        mask = _get_element_mask_from_nodes(net, tbl, [fjc, tjc], junctions)
+        if tbl not in ignore_in_service_branch_elements:
+            active_col = comp.active_identifier()
+            mask &= net[tbl][active_col].to_numpy()
+        fj = net[tbl][fjc].to_numpy()[mask]
+        tj = net[tbl][tjc].to_numpy()[mask]
+        if len(fj) == 0:
+            continue
+        if isinstance(weight_column_lookup, dict):
+            weight_col = weight_column_lookup.get(tbl, "length_km")
+        else:
+            weight_col = weight_column_lookup
+        if weight_col in net[tbl].columns:
+            weights = net[tbl][weight_col].to_numpy()[mask]
+        else:
+            weights = np.ones(len(fj)) * 0.001
+
+        g.add_edges([[pp_junction_mapping[fj], pp_junction_mapping[tj]] for (fj, tj) in zip(fj, tj)],
+                    {"weight": weights})
+
+    if additional_edge_factories is not None:
+        for tbl, edge_factory in additional_edge_factories.items():
+            edge_factory(net, g, pp_junction_mapping, junction_index)
 
     meshed = _igraph_meshed(g)
     roots = [pp_junction_mapping[s] for s in net.ext_grid.junction.values if s in junction_index]
     return g, meshed, roots  # g, (not g.is_dag())
 
 
 def create_generic_coordinates(net, mg=None, library="igraph", geodata_table="junction_geodata",
```

Comment:
As you allow kwargs, would it make sense, to pass them in functions they make use of this function:
- in create_simple_collections